rust-lib/src/lib.rs

```diff
@@ -4,10 +4,11 @@ use std::collections::HashMap;
 use std::env;
 use std::ffi::{CStr, CString};
 use std::hash::{Hash, Hasher};
-use std::os::raw::{c_char, c_void};
+use std::os::raw::{c_char, c_int, c_void};
+use std::slice;
 use std::str::FromStr;
 
-use log::{info, LevelFilter};
+use log::{info, trace, LevelFilter};
```

Comment:
We need to have
`#[macro_use] extern crate log` and remove `use log::{info, trace}`